internal_ws/yksg/src/lib.rs

```diff
@@ -45,14 +45,20 @@ impl Drop for LocalMem {
 
 impl LocalMem {
     /// Given a pointer `src` and a size, write its value to the pointer `dst`.
-    pub fn write_val(&mut self, dst: *mut u8, src: *const u8, size: usize) {
-        unsafe {
-            std::ptr::copy(src, dst, size);
-        }
+    ///
+    /// # Safety
+    ///
+    /// XXX?
+    pub unsafe fn write_val(&mut self, dst: *mut u8, src: *const u8, size: usize) {
+        std::ptr::copy(src, dst, size);
     }
 
     /// Write a constant to the pointer `dst`.
-    pub fn write_const(&mut self, dest: *mut u8, constant: &Constant) {
+    ///
+    /// # Safety
+    ///
+    /// XXX?
+    pub unsafe fn write_const(&mut self, dest: *mut u8, constant: &Constant) {
```

Comment:
I agree that both `write_val` and `write_const` should probably be unsafe, since we don't know what pointer the caller  hands it so we can't guarantee that writing to it will be safe. Even more so since `dest` could be an indirect pointer which doesn't just point into the frames memory, but could be any pointer from the user's interpreter loop.

Reply:
Can the user call these directly?!

Reply:
No, only we call them. But one of these `dest` pointers could be an `IRPlace::Indirect`, whose value is a pointer. Writing to it won't just write into our stack frame, but whereever that points, which is down to the interpreter writer, e.g. this would probably call `write_const`.
```
fn inter_step(ctx: InterpCtx) {
    ctx.val = 12;
}
```

Reply:
Just so that I'm sure: we're not doing something unsafe over and above what the user did originally?

Reply:
I sometimes get confused with the different uses of `unsafe`, but I think I can answer this with "Yes". `write_const/val` takes a pointer, which either points into the Stopgap interpreter's stack frame, or to something that was allocated in the user's interpreter, and write to it. Apart from that we are not doing anything unsafe I think.